lib/spack/spack/cmd/create.py

```diff
@@ -249,6 +257,10 @@ class RPackageTemplate(PackageTemplate):
     """Provides appropriate overrides for R extensions"""
     base_class_name = 'RPackage'
 
+    list_url = """\
+    # FIXME: Set list_url for R packages from CRAN
+    # list_url = \"https://cran.r-project.org/src/contrib/Archive/\""""
```

Comment:
If you change this to:
```suggestion
    list_url = \"https://cran.r-project.org/src/contrib/Archive/{name}\""""
```
does Spack autofill the name of the package? We may have to do a little bit of hacking because we need to preserve capitalization and other characters in the package name. You should probably add an additional `{raw_name}` or something like that. It would also be nice to only add `list_url` for RPackage and not for any other class.

Reply:
I was not able to get it to expand the name at that point in the template. What I have will work at the expense of more spidering but it is better than nothing. The old templating used to fill that out properly so it would be nice if we could get that to fill in the {name} to get back to that behavior.

I think the list_url template variable can be set to an empty string in the defaults section but I need to double check.

Reply:
@adamjstewart 
The {list_url} variable needs to be set in the defaults section to avoid an error but it can be set to an empty string. That will introduce an extra blank line in the default case. If that is preferable I can make that change.

Reply:
Regarding {name} expansion, for R it is not really the name of the package, but rather the name of the tarball, sans extension that would need to go there.

Reply:
Let me do some hacking and I'll see what I can do.

Reply:
I think it might be time to switch from simple string formatting to [Jinja2](http://jinja.pocoo.org/docs/2.10/). I've been meaning to do this for a long time, ever since @alalazo switched our module file creation to jinja2. This will allow a lot more complexity, like optionally available variables or custom package creation templates depending on your site's needs. I think what you're trying to do can still be done with simple string formatting, but there are a lot of advantages to this approach. I'm not sure when I'll find the time to do this, but I'm definitely willing to do it.